server/db/driver/pg/errors.go

```diff
@@ -0,0 +1,38 @@
+package pg
+
+import (
+	"errors"
+)
+
+// These errors are specific to the pg backend; they are not generic DEX
+// archivist errors.
+var (
+	errNoRows      = errors.New("no rows")
+	errTooManyRows = errors.New("too many rows")
+)
+
+// DetailedError pairs an Error with details.
+type DetailedError struct {
+	wrapped error
+	detail  string
+}
+
+// Error satisfies the error interface, combining the wrapped error message with
+// the details.
+func (e DetailedError) Error() string {
+	return e.wrapped.Error() + ": " + e.detail
+}
+
+// Unwrap returns the wrapped error, allowing errors.Is and errors.As to work.
+func (e DetailedError) Unwrap() error {
+	return e.wrapped
+}
+
+// NewDetailedError wraps the provided Error with details in a DetailedError,
+// facilitating the use of errors.Is and errors.As via errors.Unwrap.
+func NewDetailedError(err error, detail string) DetailedError {
+	return DetailedError{
+		wrapped: err,
+		detail:  detail,
+	}
+}
```

Comment:
@buck54321 Thoughts about this kind of error for broader usage?  There are so many places already where it would be helpful to return an error with a wrapped type that can be recognized by `errors.Is`, but for which we also want to supplement with a detailed message.  I think this accomplishes that.

Reply:
Yep. I plan to spend some time focused on error handling soon too, so that will be helpful.